src/toil/batchSystems/kubernetes.py

```diff
@@ -677,44 +677,56 @@ def getUpdatedBatchJob(self, maxWait):
-                        pod = event['object']
-                        if pod.metadata.name.startswith(self.jobPrefix):
-                            if pod.status.phase == 'Failed' or pod.status.phase == 'Succeeded':
-                                containerStatuses =  pod.status.container_statuses
-                                logger.debug("FINISHED")
-                                if containerStatuses is None or len(containerStatuses) == 0: 
-                                    logger.debug("No job container statuses for job %s" % (pod.metadata.owner_references[0].name))
-                                    return UpdatedBatchJobInfo(jobID=int(pod.metadata.owner_references[0].name[len(self.jobPrefix):]), exitStatus=EXIT_STATUS_UNAVAILABLE_VALUE, wallTime=0, exitReason=None)
-
-                                # Get termination onformation from the pod
-                                termination = pod.status.container_statuses[0].state.terminated
-                                logger.info("REASON: %s Exit Code: %s", termination.reason, termination.exit_code)
-                                
-                                if termination.exit_code != 0:
-                                    # The pod failed. Dump information about it.
-                                    logger.debug('Failed pod information: %s', str(pod))
-                                    logger.warning('Log from failed pod: %s', self._getLogForPod(pod))
-                                jobID = int(pod.metadata.owner_references[0].name[len(self.jobPrefix):])
-                                terminated = pod.status.container_statuses[0].state.terminated
-                                runtime = slow_down((terminated.finished_at - terminated.started_at).total_seconds())
-                                result = UpdatedBatchJobInfo(jobID=jobID, exitStatus=terminated.exit_code, wallTime=runtime, exitReason=None)
-                                self._try_kubernetes(self._api('batch').delete_namespaced_job, 
-                                            pod.metadata.owner_references[0].name,
+            for event in self._try_kubernetes_stream(self._api('batch').list_namespaced_job, self.namespace, 
+                                                        label_selector="toil_run={}".format(self.runID),
+                                                        timeout_seconds=maxWait):
+                # Grab the metadata data, ID, the list of conditions of the current job, and the total pods
+                jobObject = event['object']
+                jobID = int(jobObject.metadata.name[len(self.jobPrefix):])
+                jobObjectListConditions =jobObject.status.conditions
+                totalPods = jobObject.status.active + jobObject.status.finished + jobObject.status.failed
+                # Exit Reason defaults to 'Successfully Finsihed` unless said otherwise 
+                ExitReason = 1
```

Comment:
I think we want to import BatchJobExitReason and use e.g. `BatchJobExitReason.FINISHED` instead of just `1`, in case we reassign the numbers, and so we can tell what they mean.

Also, variables probably shouldn't be capitalized like this; Github is highlighting them like classes.